src/network/getters/getSponsorshipStats.ts

```diff
@@ -0,0 +1,109 @@
+import moment from 'moment'
+import { ChartPeriod } from '~/shared/components/NetworkChart/NetworkChart'
+import { GetSponsorshipDailyBucketsQuery } from '~/generated/gql/network'
+import { toBN } from '~/utils/bn'
+import { getSponsorshipDailyBuckets } from './getSponsorshipDailyBuckets'
+import { getSponsorshipTokenInfo } from './getSponsorshipTokenInfo'
+
+export const getSponsorshipStats = async (
+    sponsorshipId: string,
+    selectedPeriod: ChartPeriod,
+    dataSource: string,
+    ignoreToday?: boolean,
+): Promise<{ x: number; y: number }[]> => {
+    const tokenInfo = await getSponsorshipTokenInfo()
+    const today = moment()
+    const start = ignoreToday ? moment().utc().startOf('day') : moment().utc()
+
+    let result: GetSponsorshipDailyBucketsQuery['sponsorshipDailyBuckets']
+    switch (selectedPeriod) {
+        case ChartPeriod.SevenDays:
+            result = await getSponsorshipDailyBuckets(
+                sponsorshipId,
+                start.format('X'),
+                start.clone().subtract(7, 'days').format('X'),
+            )
+            break
+        case ChartPeriod.OneMonth:
+            result = await getSponsorshipDailyBuckets(
+                sponsorshipId,
+                start.format('X'),
+                start.clone().subtract(30, 'days').format('X'),
+            )
+            break
+        case ChartPeriod.ThreeMonths:
+            result = await getSponsorshipDailyBuckets(
+                sponsorshipId,
+                start.format('X'),
+                start.clone().subtract(90, 'days').format('X'),
+            )
+            break
+        case ChartPeriod.OneYear:
+            result = await getSponsorshipDailyBuckets(
+                sponsorshipId,
+                start.format('X'),
+                start.clone().subtract(365, 'days').format('X'),
+            )
+            break
+        case ChartPeriod.YearToDate:
+            const beginningOfYear = moment().utc().startOf('year')
+            const daySpan = today.clone().utc().diff(beginningOfYear, 'day')
+            if (daySpan === 0) {
+                result = []
+                break
+            }
+            result = await getSponsorshipDailyBuckets(
+                sponsorshipId,
+                start.format('X'),
+                start.clone().subtract(daySpan, 'days').format('X'),
+            )
+            break
+        case ChartPeriod.All:
+            const maxAmount = 999
+            const maxIterations = 5
+            const endDate = start.clone().subtract(maxIterations * maxAmount, 'days')
+            const elements: GetSponsorshipDailyBucketsQuery['sponsorshipDailyBuckets'] =
+                []
+            // yeah - I'm guessing we will not have a history longer than 5 thousand days :)
+            for (let i = 0; i < maxIterations; i++) {
+                const partialResult = await getSponsorshipDailyBuckets(
+                    sponsorshipId,
+                    start.format('X'),
```

Comment:
If you iterate, shouldn't start also be affected by the current window? I.e. 

```
0: start0 → end0,
1: start1 → end1,
…
N: startN → endN
```

```ts
// [start0, end0], [start1, end1]
```

Reply:
no, it's being handled by the 'skip' parameter of the query. so it's like the date filters define the whole range and 'first' & 'skip' params take care of the iteration

Reply:
Ah right! Wp.